cluster-autoscaler/loop/trigger.go

```diff
@@ -74,16 +97,23 @@ func (t *LoopTrigger) Wait(lastRun time.Time) {
 		default:
 			klog.Infof("Autoscaler loop triggered immediately after a productive iteration")
 		}
-		return
-	}
-
-	// Unschedulable pod triggers autoscaling immediately.
-	select {
-	case <-time.After(t.scanInterval):
-		klog.Infof("Autoscaler loop triggered by a %v timer", t.scanInterval)
-	case <-t.podObserver.unschedulablePodChan:
-		klog.Info("Autoscaler loop triggered by unschedulable pod appearing")
+	} else if t.provisioningRequestProcessTimeGetter != nil && !t.provisioningRequestProcessTimeGetter.LastProvisioningRequestProcessTime().Before(lastRun) {
+		select {
+		case <-t.podObserver.unschedulablePodChan:
+			klog.Info("Autoscaler loop triggered by unschedulable pod appearing")
+		default:
+			klog.Infof("Autoscaler loop triggered immediately after a provisioning request was processed")
+		}
+	} else {
```

Comment:
Pls. revert to removing the "else" branch (this is the preferred convention in golang, see https://go.dev/doc/effective_go#control-structures).

Reply:
Updated